### ISSUE TITLE
[adjacent-overload-signatures] False positive on call signatures and a method named `call`

### ISSUE THREAD

**Issue by magurotuna** [typescript-eslint/typescript-eslint#2312]
**Repro**

```js
module.exports = {
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint"],
  rules: {
    "@typescript-eslint/adjacent-overload-signatures": "error",
  },
};

```

```TS
interface Foo {
  (): void;
  n: number;
  call(): void;
}
```

**Expected Result**
No lint error is detected

**Actual Result**
```
  4:3  error  All 'call' signatures should be adjacent  @typescript-eslint/adjacent-overload-signatures

✖ 1 problem (1 error, 0 warnings)

```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.1` |
| `@typescript-eslint/parser`        | `3.6.1` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `12.13.1` |
| `npm`                              | `6.12.1` |
